website/docs/plugin/framework/ephemeral-resources/close.mdx

```diff
@@ -0,0 +1,94 @@
+---
+page_title: 'Plugin Development - Framework: Open Ephemeral Resources'
+description: >-
+  How to implement ephemeral resource close in the provider development framework.
+---
+
+# Close Ephemeral Resources
+
+Close is an optional part of the Terraform lifecycle for an ephemeral resource, which is different from the [managed resource lifecycle](https://github.com/hashicorp/terraform/blob/main/docs/resource-instance-change-lifecycle.md). During any Terraform operation (like [`terraform plan`](/terraform/cli/commands/plan) or [`terraform apply`](/terraform/cli/commands/apply)), when an ephemeral resource's data is needed, Terraform initially retrieves that data with the [`Open`](/terraform/plugin/framework/ephemeral-resources/open) lifecycle handler. Once the ephemeral resource data is no longer needed, Terraform calls the provider `CloseEphemeralResource` RPC, in which the framework calls the [`ephemeral.EphemeralResourceWithClose` interface `Close` method](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/ephemeral#EphemeralResourceWithClose). The request contains any `Private` data set in the latest `Open` or `Renew` call.
+
+`Close` is an optional lifecycle implementation for an ephemeral resource, other lifecycle implementations include:
+
+- [Open](/terraform/plugin/framework/ephemeral-resources/open) an ephemeral resource by receiving Terraform configuration, retrieving a remote object, and returning ephemeral result data to Terraform.
+- [Renew](/terraform/plugin/framework/ephemeral-resources/renew) an expired remote object at a specified time.
+
+## Define Close Method
+
+The [`ephemeral.EphemeralResourceWithClose` interface](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/ephemeral#EphemeralResourceWithClose) on the [`ephemeral.EphemeralResource` interface](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/ephemeral#EphemeralResource) implementation will enable close support for an ephemeral resource.
+
+Implement the `Close` method by:
+
+1. [Accessing private data](/terraform/plugin/framework/resources/private-state#reading-private-state-data) from [`ephemeral.CloseRequest.Private` field](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/ephemeral#CloseRequest.Private) needed to close the remote object.
```

Comment:
Yeah we should either make this page more generic and move it, or just duplicate it. I just opted to delay that for now since it gets the general information across. I think we can come back and adjust it later 👍🏻